cpp/src/prims/per_v_pair_transform_dst_nbr_intersection.cuh

```diff
@@ -0,0 +1,357 @@
+ * @tparam EdgeDstValueInputWrapper Type of the wrapper for edge destination property values.
+ * @tparam IntersectionOp Type of the quinary per intersection operator.
+ * @tparam VertexPairValueOutputIterator Type of the iterator for vertex pair output property
+ * variables.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param vertex_pair_first Iterator pointing to the first (inclusive) input vertex pair.
+ * @param vertex_pair_last Iterator pointing to the last (exclusive) input vertex pair.
+ * @param edge_src_value_input Wrapper used to access source input property values (for the edge
+ * sources assigned to this process in multi-GPU). Use either cugraph::edge_src_property_t::view()
+ * (if @p e_op needs to access source property values) or cugraph::edge_src_dummy_property_t::view()
+ * (if @p e_op does not access source property values). Use update_edge_src_property to fill the
+ * wrapper.
+ * @param edge_dst_value_input Wrapper used to access destination input property values (for the
+ * edge destinations assigned to this process in multi-GPU). Use either
+ * cugraph::edge_dst_property_t::view() (if @p e_op needs to access destination property values) or
+ * cugraph::edge_dst_dummy_property_t::view() (if @p e_op does not access destination property
+ * values). Use update_edge_dst_property to fill the wrapper.
+ * @param intersection_op quinary operator takes first vertex of the pair, second vertex of the
+ * pair, property values for the first vertex, property values for the second vertex, and a list of
+ * vertices in the intersection of the first & second vertices' destination neighbors and returns an
+ * output value for the input pair.
+ * @param vertex_pair_value_output_first Iterator pointing to the vertex pair property variables for
+ * the first vertex pair (inclusive). `vertex_pair_value_output_last` (exclusive) is deduced as @p
+ * vertex_pair_value_output_first + @p thrust::distance(vertex_pair_first, vertex_pair_last).
+ * @param A flag to run expensive checks for input arguments (if set to `true`).
+ */
+template <typename GraphViewType,
+          typename VertexPairIterator,
+          typename EdgeSrcValueInputWrapper,
+          typename EdgeDstValueInputWrapper,
+          typename IntersectionOp,
+          typename VertexPairValueOutputIterator>
+void per_v_pair_transform_dst_nbr_intersection(
+  raft::handle_t const& handle,
+  GraphViewType const& graph_view,
+  VertexPairIterator vertex_pair_first,
+  VertexPairIterator vertex_pair_last,
+  EdgeSrcValueInputWrapper edge_src_value_input,
+  EdgeDstValueInputWrapper edge_dst_value_input,
+  IntersectionOp intersection_op,
+  VertexPairValueOutputIterator vertex_pair_value_output_first,
+  bool do_expensive_check = false)
```

Comment:
@ChuckHastings Still a Work-in-Progress, but you may start integrating this to your Jaccard/Sorensen/Overlap coefficients code.